traffic_ops/app/lib/TrafficOpsRoutes.pm

```diff
@@ -657,6 +657,23 @@ sub api_routes {
 	# Supports ?orderby=key
 	$r->get("/api/$version/roles")->over( authenticated => 1 )->to( 'Role#index', namespace => $namespace );
 
+	# -- CAPABILITIES
+	# Supports ?orderby=key
+	$r->get("/api/$version/capabilities")->over( authenticated => 1 )->to( 'Capability#all', namespace => $namespace );
+	$r->get("/api/$version/capabilities/:name")->over( authenticated => 1 )->to( 'Capability#name', namespace => $namespace );
+	$r->put("/api/$version/capabilities/:name")->over( authenticated => 1 )->to( 'Capability#update', namespace => $namespace );
+	$r->post("/api/$version/capabilities")->over( authenticated => 1 )->to( 'Capability#create', namespace => $namespace );
+	$r->delete("/api/$version/capabilities/:name")->over( authenticated => 1 )->to( 'Capability#delete', namespace => $namespace );
+
+	# -- API-CAPABILITIES
+	# Supports ?orderby=key
+	$r->get("/api/$version/api_capabilities")->over( authenticated => 1 )->to( 'ApiCapability#all', namespace => $namespace );
+	$r->get("/api/$version/api_capabilities/:id")->over( authenticated => 1 )->to( 'ApiCapability#index', namespace => $namespace );
+	$r->get("/api/$version/api_capabilities/capability/:name")->over( authenticated => 1 )->to( 'ApiCapability#capName', namespace => $namespace );
```

Comment:
so we typically use query params to filter data. so i don't think this is needed:

$r->get("/api/$version/api_capabilities/capability/:name

instead,  you can just do GET api/*/api_capabilities?capability=[whatever] but just make sure your "index" action can handle the query param to filter the result set.

Here's an example where we use a query param to filter regions by division:

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Region.pm#L28